### PR TITLE
Rename and expose inputs_owner function

### PR DESCRIPTION
**Pull request by SwayStar123**
## Description
Renamed the `inputs_owner` function to `caller_address` to be consistent with the `caller_contract_id` function, and makes the function public, changing its return type from a Result of an Identity to a Result of an Address

## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-lib-std/src/auth.sw [hunk 1]
```diff
@@ -79,7 +81,8 @@ pub fn caller_contract_id() -> ContractId {
 ///
 /// # Additional Information
 ///
-/// Returns a Err if the caller is external and the inputs to the transaction are not all owned by the same address.
+/// Returns a AuthError::InputsNotAllOwnedBySameAddress if the caller is external and the inputs to the transaction are not all owned by the same address.
```
Comment (ironcev) [inline sway-lib-std/src/auth.sw:84]:
```suggestion
/// Returns an `AuthError::InputsNotAllOwnedBySameAddress` if the caller is external and the inputs to the transaction are not all owned by the same address.
```

File: sway-lib-std/src/auth.sw [hunk 2]
```diff
@@ -79,7 +81,8 @@ pub fn caller_contract_id() -> ContractId {
...
+/// Should not return a AuthError::CallerIsInternal under any circumstances.
```
Comment (ironcev) [inline sway-lib-std/src/auth.sw:85]:
```suggestion
/// Should not return an `AuthError::CallerIsInternal` under any circumstances.
```

**Review by ironcev — APPROVED** [review]
Two slight details in the docs otherwise LGTM

**Review by ironcev — DISMISSED** [review]

File: sway-lib-std/src/auth.sw [hunk 3]
```diff
@@ -93,12 +96,17 @@ pub fn caller_contract_id() -> ContractId {
 ///         Ok(Identity::Address(address)) => log(address),
 ///         Ok(Identity::ContractId(contract_id)) => log(contract_id),
 ///         Err(AuthError::InputsNotAllOwnedBySameAddress) => log("Inputs not all owned by same address."),
+///         Err(AuthError::CallerIsInternal) => log("Hell froze over."),
```
Comment (K1-R1) [inline sway-lib-std/src/auth.sw:99]: 😆

File: sway-lib-std/src/auth.sw [hunk 4]
```diff
@@ -93,12 +96,17 @@ pub fn caller_contract_id() -> ContractId {
...
 ///     }
 /// }
 /// ```
 pub fn msg_sender() -> Result<Identity, AuthError> {
     if caller_is_external() {
-        inputs_owner()
+        let inputs_res = caller_address();
```
Comment (K1-R1) [inline sway-lib-std/src/auth.sw:105]:
```suggestion
        match caller_address() {
            Err(err) => Err(err),
            Ok(owner) => Ok(Identity::Address(owner)),
        }
```

File: sway-lib-std/src/auth.sw [hunk 5]
```diff
@@ -169,6 +172,9 @@ fn inputs_owner() -> Result<Identity, AuthError> {
         return Err(AuthError::InputsNotAllOwnedBySameAddress);
     }
 
-    // `candidate` must be `Some` at this point, so can unwrap safely.
-    Ok(Identity::Address(candidate.unwrap()))
+    // `candidate` must be `Some` if the caller is an Address, Contract otherwise.
```
Comment (K1-R1) [inline sway-lib-std/src/auth.sw:175]:
```suggestion
    // `candidate` must be `Some` if the caller is an Address, otherwise it's a contract.
```

**Review by K1-R1 — CHANGES_REQUESTED** [review]
A few changes requested; but it looks good

**Review by ironcev — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**